messaging/src/main/java/org/axonframework/common/jdbc/PagingJdbcIterable.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.common.jdbc;
+
+import org.axonframework.common.transaction.TransactionManager;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static org.axonframework.common.jdbc.JdbcUtils.executeQuery;
+import static org.axonframework.common.jdbc.JdbcUtils.listResults;
+
+/**
+ * Enables iterating through a JDBC query using paging. Paging is taken care of automatically through the provided
+ * {@link PagingStatementSupplier}, fetching the next page when the items run out to iterate through.
+ * <p>
+ * Do not use this for paging when you care about concurrent deletes. If you loaded a page, delete an item from it, and
+ * load the next, you will miss an item during iteration.
+ * <p>
+ * The {@link #iterator()} function can be called multiple times to loop through the items, restarting the query from
+ * the start.
+ *
+ * @param <R> The mapped result type.
+ * @author Mitchell Herrijgers
+ * @author Steven van Beelen
```

Comment:
Author credits? :o I'm honored

Reply:
It's based on the `PagingJpaQueryIterable`, so I figured it was only fair to give you credits too. :-)